typings/styled-components.d.ts

```diff
@@ -0,0 +1,162 @@
+import * as React from "react";
+import { StatelessComponent } from "react";
+
+type ConstrainedProps<C, P> = C & ({ defaultProps?: P } | { new(props?: P, context?: any): any });
+type StyledProps<P> = P & { theme: any };
+
+interface StyledFunction<T, P> {
+  (strs: TemplateStringsArray, ...fns: Array<(props: P) => string>): T;
```

Comment:
Ah, @patrick91 you already filed such issue :) https://github.com/Microsoft/TypeScript/issues/11947 sorry